### PR TITLE
Feature/616

### LINKED ISSUE #616: Automatic selector populating for diamond facets
We have 2 files where we [add selectors](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/test/diamond/DiamondTestSetup.sol#L138) for the diamond storage:
1. https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol
2. https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/test/diamond/DiamondTestSetup.sol

We add all the selectors manually so it is easy to miss one or forget to update the migration script after changes to one of the contracts.

What should be done:
1. Automate selector populating both in these files: [01_Diamond](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol) and [DiamondTestSetup](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/test/diamond/DiamondTestSetup.sol). Foundry can read from json files so perhaps we could read the build ABIs and automatically populate selectors for all facets.
2. Use a dynamic array [here](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol#L140) and [here](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/test/diamond/DiamondTestSetup.sol#L538)

### PR DESCRIPTION
**Pull request by rndquu**
Resolves https://github.com/ubiquity/ubiquity-dollar/issues/616

### CODE REVIEW

**Comment by rndquu** [comment]
@molecula451 I have no idea how the "Compare Test Coverage" workflow calculates the following values and fails afterwards:
```
Development branch coverage: 69.1
PR branch coverage: 10.1
```

I will raise a bug in a separate issue.

**Comment by molecula451** [comment]
> @molecula451 I have no idea how the "Compare Test Coverage" workflow calculates the following values and fails afterwards:
> 
> ```
> Development branch coverage: 69.1
> PR branch coverage: 10.1
> ```
> 
> I will raise a bug in a separate issue.

i think it will fine to have a branch coverage issue as this architecture in the workflow looks a bit outdated, leading to this failing compared test coverages

File: .github/workflows/coverage-check.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
           ref: development
 
       - name: Setup Foundry
-        uses: onbjerg/foundry-toolchain@v1
+        uses: foundry-rs/foundry-toolchain@v1
```
Comment (molecula451) [inline .github/workflows/coverage-check.yml:24]: super important catch

File: packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol [hunk 2]
```diff
@@ -272,266 +271,50 @@ contract DiamondScript is Constants {
...
+            "/out/StakingFormulasFacet.sol/StakingFormulasFacet.json"
         );
-        // Dollar Mint Excess
-        selectorsOfDollarMintExcessFacet.push(
-            (dollarMintExcessFacet.distributeDollars.selector)
+        selectorsOfTWAPOracleDollar3poolFacet = getSelectorsFromAbi(
+            "/out/TWAPOracleDollar3poolFacet.sol/TWAPOracleDollar3poolFacet.json"
         );
 
         // Credit Clock Facet
```
Comment (molecula451) [inline packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol]: @rndquu looks like you're leaving out the recent facet selectors

File: .github/workflows/coverage-check.yml (hunk 1 above)
Comment (rndquu) [reply .github/workflows/coverage-check.yml:24]: it's not :)

File: packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol (hunk 2 above)
Comment (rndquu) [reply packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol]: fixed

**Review by molecula451 — APPROVED** [review]
litte script to fetch the facets selectors from facet working nice, overall good code

**Pull request merged**